test_conformance/compiler/test_feature_macro.cpp

```diff
@@ -0,0 +1,398 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "testBase.h"
+#include <vector>
+#include <algorithm>
+
+const char* macro_supported_source =
+    "kernel void enabled(global int * buf) { \r\n"
+    "int n = get_global_id(0); \r\n"
+    "buf[n] = 0; \r\n "
+    "#ifndef %s \r\n"
+    "ERROR; \r\n"
+    "#endif \r\n"
+    "\r\n } \r\n";
+
+
+const char* macro_not_supported_source =
+    "kernel void not_enabled(global int * buf) { \r\n"
+    "int n = get_global_id(0); \r\n"
+    "buf[n] = 0; \r\n "
+    "#ifdef %s \r\n"
+    "ERROR; \r\n"
+    "#endif \r\n"
+    "\r\n } \r\n";
+
+
+template <typename T>
+cl_int check_api_feature_info(cl_device_id deviceID, cl_context context,
```

Comment:
This should ideally be split into separate functions to deal with the different kinds of queries. You could have several specialised `test_feature_macro` functions (overloads or different names) that share the compiler and comparison part. This is not strictly required but would make the code a bit easier to read IMHO.